File: src/main/java/io/xol/chunkstories/api/entity/DamageCause.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.entity;

public interface DamageCause {
	public String getName();

	/**
	 * How many milliseconds should the target be set invulnerable after an attack
	 */
	public default long getCooldownInMs() {
		return 0;
	}

	public static DamageCause DAMAGE_CAUSE_FALL = new DamageCause() {

		@Override
		public String getName() {
			return "damage.fall";
		}

	};
}

File: src/main/java/xyz/chunkstories/api/entity/DamageCause.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.entity

interface DamageCause {
    val name: String

    /** How many milliseconds should the target be set invulnerable after an
     * attack  */
    val cooldownInMs: Long
        get() = 0

    companion object {

        val DAMAGE_CAUSE_FALL: DamageCause = object : DamageCause {

            override val name: String
                get() = "damage.fall"

        }
    }
}
